src/training/character_specific/ptrainer.rs

```diff
@@ -0,0 +1,160 @@
+    if status_kind == *WEAPON_PTRAINER_PTRAINER_STATUS_KIND_RUN {
+        MotionModule::set_rate(ptrainer_module_accessor, 1.0);
+    }
+    if frame_counter::should_delay(5_u32, *SWITCH_DELAY_COUNTER) {
+        // Need to wait to make sure we stop the flash effect
+        return false;
+    }
+    // If you're trying to fix PT getting locked up, maybe try
+    //  to run FUN_71000106c0 in lua2cpp_ptrainer to get her wandering correct
+    // Also worth trying is figuring out how to prevent PT from entering WEAPON_PTRAINER_PTRAINER_STATUS_KIND_RUN_STOP
+    //  instead of run below after that status change
+    if situ_kind == SITUATION_KIND_AIR {
+        StatusModule::set_situation_kind(
+            ptrainer_module_accessor,
+            app::SituationKind(*SITUATION_KIND_GROUND),
+            true,
+        );
+        StatusModule::change_status_force(
+            ptrainer_module_accessor,
+            *WEAPON_PTRAINER_PTRAINER_STATUS_KIND_RUN,
+            false,
+        );
+    }
+    true
+}
+
+pub unsafe fn change_motion(
+    ptrainer_module_accessor: &mut app::BattleObjectModuleAccessor,
+    motion_kind: u64,
+) {
+    if app::utility::get_kind(ptrainer_module_accessor) == *WEAPON_KIND_PTRAINER_PTRAINER
+        && hash40("restart") == motion_kind
+        && save_states::is_loading()
+    {
+        MotionModule::set_rate(ptrainer_module_accessor, 1000.0);
+    }
+}
+
+pub unsafe fn get_ptrainer_mball_module_accessor(
+    ptrainer_module_accessor: &mut app::BattleObjectModuleAccessor,
+) -> Option<&mut app::BattleObjectModuleAccessor> {
+    if ArticleModule::is_exist(
+        ptrainer_module_accessor,
+        *WEAPON_PTRAINER_PTRAINER_GENERATE_ARTICLE_MBALL,
+    ) {
+        let ptrainer_masterball: *mut app::Article = ArticleModule::get_article(
+            ptrainer_module_accessor,
+            *WEAPON_PTRAINER_PTRAINER_GENERATE_ARTICLE_MBALL,
+        );
+        let ptrainer_masterball_id = Article::get_battle_object_id(ptrainer_masterball);
+        return Some(&mut *app::sv_battle_object::module_accessor(
+            ptrainer_masterball_id as u32,
+        ));
+    }
+    None
+}
+
+pub unsafe fn get_ptrainer_module_accessor(
+    module_accessor: &mut app::BattleObjectModuleAccessor,
+) -> &mut app::BattleObjectModuleAccessor {
+    let ptrainer_object_id =
+        LinkModule::get_parent_object_id(module_accessor, *FIGHTER_POKEMON_LINK_NO_PTRAINER);
+    &mut *app::sv_battle_object::module_accessor(ptrainer_object_id as u32)
+}
+
+pub unsafe fn get_pokemon_module_accessor(
+    ptrainer_module_accessor: *mut app::BattleObjectModuleAccessor,
+) -> *mut app::BattleObjectModuleAccessor {
+    let pokemon_object_id = LinkModule::get_node_object_id(
+        ptrainer_module_accessor,
+        *WEAPON_PTRAINER_PTRAINER_LINK_NO_POKEMON,
+    );
+    &mut *app::sv_battle_object::module_accessor(pokemon_object_id as u32)
+}
+
+pub unsafe fn check_effect_pokemon_state(
```

Comment:
nit: This isn't really a `check`. Usually if something can return something to potentially mutate a hook, we call it `handle`, and also it's about the size. So maybe `handle_pokemon_effect_size`

Reply:
I prefer `handle_pokemon_effect`, but agree with the sentiment here. It's not really about resizing the effect, and more that that is the easiest way to "remove" it.